src/main/java/tc/oc/pgm/map/Contributor.java

```diff
@@ -103,4 +113,37 @@ public boolean hasContribution() {
     }
     return resolved;
   }
+
+  public static boolean isUUIDCached(UUID uuid) {
+    return cachedNames.containsKey(uuid);
+  }
+
+  public static String getCachedName(UUID uuid) {
+    return cachedNames.get(uuid);
+  }
+
+  public static void cacheName(UUID uuid, String name) {
+    cachedNames.put(uuid, name);
+  }
+
+  public static void resolve(UUID uuid) throws IOException {
```

Comment:
The Contributor class shouldn't be handling this logic, move it to a name resolver or name cache util